### PR TITLE
Feature db api impl

### PR DESCRIPTION
**Pull request by User12837985123**
### Feature Description
Implements the server side of the API, solves [this trello task](https://trello.com/c/Ij0BeKFI/214-db-api-endpoints). Documentation and access to the server deployed endpoints can be found at http://130.225.57.95:8393/swagger/. Note that we also removed the hardcoded username and password for mongo admin access (this is why you have to manually add mongoConnection.json).

### Most Relevant Files - (Review to the best of your ability)
aspnet-server1/src/IO.Swagger/Controllers/DeadCrumpsApi.cs
aspnet-server-1/MongoConnectionService.cs

### How To Run
Compiling and running the API requires ASP.NET Core
The API accesses the server database, so be mindful of what you post :D
#### Locally
1. Place the config file names mongoConnection.json from Discord in the folder P8/aspnet-server-1/.
2. If Ubuntu
2.1 run "sh build.sh"
3. If Windows cmd (Alternatively you can also build it through Visual Studio)
3.1 run "build.bat"
4. then run the command it tells you to run, something like "dotnet run -p src/IO.Swagger/IO.Swagger.csproj --launch-profile web"
5. Then go to the url it says it is running on. (localhost:8393)

### How we did setup on the server
1. Login to the server via ssh (see IP in Discord)
2. The newest version of the API should be on the server, otherwise, pull the branch.
3. run sudo sh build.sh
4. Then run the command it tells you to run, something like "dotnet run -p src/IO.Swagger/IO.Swagger.csproj --launch-profile web
5. Then try an access it on port 8393

This screenshot shows what the mongo db contains at the time of creating this pull request. This mock data should be fairly accurate to what we expect from our application. Note that we choose to make the username the primary key for users and that a location has a reference to the associated user.
image

### CODE REVIEW

**Review by Bjerkk — DISMISSED** [review]

File: aspnet-server-1/src/IO.Swagger/wwwroot/swagger-original.json [hunk 1]
```diff
@@ -1,104 +1,96 @@
...
+              "application/json" : {
+                "schema" : {
+                  "$ref" : "#/components/schemas/location"
+                }
+              }
+            }
           },
           "400" : {
-            "description" : "deletion of location went wrong"
+            "description" : "retrival of location went wrong"
```
Comment (Sworgon) [inline aspnet-server-1/src/IO.Swagger/wwwroot/swagger-original.json]: retrieval*

File: client-api-spec.yaml [hunk 2]
```diff
@@ -1,127 +1,132 @@
...
     post:
       summary: adds a new location
       operationId: postLocation
       requestBody:
-        $ref: '#/components/requestBodies/Location'
+        $ref: '#/components/requestBodies/location'
       responses:
         201:
-          description: location added successfully
+          description: Succesfull request! Created new location
```
Comment (Sworgon) [inline client-api-spec.yaml:66]: successful*

File: aspnet-server-1/src/IO.Swagger/Controllers/DeadCrumbsApi.cs [hunk 3]
```diff
@@ -0,0 +1,167 @@
...
+        /// adds a new location
+        /// </summary>
+        /// <param name="location">a JSON object of a location</param>
+        /// <response code="201">Succesfull request! Created new location</response>
+        /// <response code="400">Bad request! Location not added</response>
+        [HttpPost]
+        [Route("/Location")]
+        [ValidateModelState]
+        [SwaggerOperation("PostLocation")]
+        public virtual IActionResult PostLocation([FromBody]Location location)
```
Comment (Sworgon) [inline aspnet-server-1/src/IO.Swagger/Controllers/DeadCrumbsApi.cs:124]: According to Swagger, the response code can be 200 which is inconsistent with this function.

File: aspnet-server-1/src/IO.Swagger/Controllers/DeadCrumbsApi.cs [hunk 4]
```diff
@@ -0,0 +1,167 @@
...
+        }
+
+
+        /// <summary>
+        /// adds a new user
+        /// </summary>
+        /// <param name="user">a JSON object of a location</param>
+        /// <response code="201">Succesfull request! Created new user</response>
+        /// <response code="400">Bad request! User not added</response>
+        [HttpPost]
```
Comment (Sworgon) [inline aspnet-server-1/src/IO.Swagger/Controllers/DeadCrumbsApi.cs:149]: Same as above

**Review by Sworgon — DISMISSED** [review]
It looks nice, but it would be better if it had more comments

File: aspnet-server-1/src/IO.Swagger/wwwroot/swagger-original.json (hunk 1 above)
Comment (KVISDAOWNER) [reply aspnet-server-1/src/IO.Swagger/wwwroot/swagger-original.json]: Fixed

File: client-api-spec.yaml (hunk 2 above)
Comment (KVISDAOWNER) [reply client-api-spec.yaml:66]: Fixed

File: aspnet-server-1/src/IO.Swagger/Controllers/DeadCrumbsApi.cs (hunk 3 above)
Comment (KVISDAOWNER) [reply aspnet-server-1/src/IO.Swagger/Controllers/DeadCrumbsApi.cs:124]: Should be fixed

File: aspnet-server-1/src/IO.Swagger/Controllers/DeadCrumbsApi.cs (hunk 4 above)
Comment (KVISDAOWNER) [reply aspnet-server-1/src/IO.Swagger/Controllers/DeadCrumbsApi.cs:149]: Should be fixed

**Review by Sworgon — APPROVED** [review]
I like it but I would prefer more comments

**Review by Bjerkk — APPROVED** [review]

**Pull request merged**